File: 1578 minimum-time-to-make-rope-colorful/solution.cpp
class Solution {
public:
    int minCost(string s, vector<int>& neededTime) {
        int l = 0;
        for(int i=1;i<neededTime.size();i++) {
            if(s[i]==s[i-1]) {
                l+=min(neededTime[i-1],neededTime[i]);
                neededTime[i] = max(neededTime[i],neededTime[i-1]);
            }
        }
        return l;
    }
};

File: 1578 minimum-time-to-make-rope-colorful/solution.cs
public class Solution {
    public int MinCost(string s, int[] neededTime) {
        int l = 0;
        for(int i=1;i<neededTime.Length;i++) {
            if(s[i]==s[i-1]) {
                l+=Math.Min(neededTime[i-1],neededTime[i]);
                neededTime[i] = Math.Max(neededTime[i],neededTime[i-1]);
            }
        }
        return l;
    }
}
